scripts/generate_activity_script/src/script.ts

```diff
@@ -0,0 +1,353 @@
+   */
+  let factory_balance: Coin = await neutronWallet.clientCW.getBalance(
+    address,
+    FACTORY_DENOM
+  );
+  if (Number(factory_balance.amount) === 0) {
+    return null;
+  }
+
+  /* If any error occured when executing method then just ignore
+   * It's content and return null, script will try to call another method
+   */
+  let random_amount: number = Math.floor(
+    Math.random() * Number(factory_balance.amount) + 1
+  );
+  try {
+    const res = await unbond(drop_instance, address, {
+      amount: String(random_amount),
+      denom: FACTORY_DENOM,
+    });
+    if ((await neutronWallet.clientCW.getTx(res.txHash)).code !== 0) {
+      return null;
+    }
+    return res;
+  } catch (e) {
+    return null;
+  }
+}
+
+async function send_nft(
+  withdrawal_voucher: DropWithdrawalVoucher,
+  withdrawal_manager: DropWithdrawalManager,
+  address: string,
+  nft_id: string
+): Promise<Action> {
+  /* To withdraw unbonded amount we need to send nft to the withdrawal_manager contract
+   * To do that we need to call send method on withdrawal_voucher contract (which is NFT contract underhood)
+   * Field msg here is encoded base64 json object { "withdraw": {} }
+   */
+  return {
+    mode: MODE.WITHDRAW,
+    txHash: (
+      await withdrawal_voucher.sendNft(
+        address,
+        {
+          contract: withdrawal_manager.contractAddress,
+          token_id: nft_id,
+          msg: "eyJ3aXRoZHJhdyI6e319",
+        },
+        "auto",
+        "",
+        []
+      )
+    ).transactionHash,
+  };
+}
+
+/* Function returns ActionLog if transaction succeeded and null if any error or problem occurred
+ * We don't really need to know type of error here since it isn't crucial for us,
+ * We'll just try another method in core
+ */
+async function withdraw_random_nft(
+  neutronWallet: Wallet,
+  drop_instance: DropCoreClient
+): Promise<Action | null> {
+  const address: string = neutronWallet.mainAccounts[0].address;
+  /* Get both withdrawal_manager and withdrawal_voucher wrappers based on querying config method
+   * We need them to execute send_nft method on withdrawal_voucher with withdrawal_manager as the recepient
+   * To withdraw our unbonded tokens we need to work with withdrawal_voucher' send_nft method
+   */
+  const config = await drop_instance.queryConfig();
+  const withdrawal_manager: DropWithdrawalManager = new DropWithdrawalManager(
+    neutronWallet.clientCW,
+    config.withdrawal_manager_contract
+  );
+  const withdrawal_voucher: DropWithdrawalVoucher = new DropWithdrawalVoucher(
+    neutronWallet.clientCW,
+    config.withdrawal_voucher_contract
+  );
+
+  /* Get all NFT tokens in 1-st loop possessed by given address
+   * Get details for each gotten NFT in 2-nd loop
+   * Check batch status for each batch_id, filter who hasn't withdrawed yet in 3-rd loop
+   */
+  let owned_nfts = (await withdrawal_voucher.queryTokens({ owner: address }))
+    .tokens;
+  let current_nft_list = owned_nfts;
```

Comment:
It's better to build this list in one `while (true)` clause. On each step, call `withdrawal_voucher.queryTokens` and add its results to the list while results are not empty

Reply:
But this code's doing exactly this tho

Reply:
```suggestion
        const ownedNfts = [];
      while (true) {
        startAfter =
          ownedNfts.length > 0 ? ownedNfts[ownedNfts.length - 1] : undefined;
        const { tokens } = await withdrawalVoucher.queryTokens({
          owner: address,
          start_after: startAfter,
        });

        if (tokens.length === 0) break;

        ownedNfts.concat(tokens);
      }
```